inc/Ui/Editor.php

```diff
@@ -187,7 +188,7 @@ public function show()
      *
      * the default action for EDIT_FORM_ADDTEXTAREA
      *
-     * @param mixed[] $data
+     * @param array{wr: bool, media_manager: bool, target: string, intro_locale: string, form: Form} $data
```

Comment:
This phpdocs does also not work, but gives more insight in the entries of the array.
/** @var dokuwiki\Form\Form $data['form'] */ is also not recognized.

Reply:
How about this:

```suggestion
     * @param array $data {wr: bool, media_manager: bool, target: string, intro_locale: string, form: Form}
```

Reply:
Bit searching: It seems Phpstorm/IntelliJ support it, but it should probably be marked with `@psalm-param`. Sounds not so practical yet.

https://suckup.de/2020/02/modern-phpdoc-annotations/

The `array{…}` notation is already recognized as array, so the split you propose is not needed, but I will check.

Reply:
if `@param array{wr: bool, media_manager: bool, target: string, intro_locale: string, form: Form} $data` is used, it is already recognized as array parameter. But the entries are not default recognized.
If I install PHPStan plugin in IDEA IntelliJ the entire line is recognized. Is this worth to recommend the use of the PHPStan plugin?

Reply:
my suggestion was not meant to be recognized by IntelliJ but by Humans. For IntelliJ marking the parameter as an array is good enough IMHO. If the phpstan syntax still works with vanilla IntelliJ I'm happy to have it, otherwise I would prefer to stay at something that works out of the box.

Reply:
Clear. Array is recognized without the PHPstan plugin.